src/coreclr/vm/ecall.cpp

```diff
@@ -336,11 +336,10 @@ PCODE ECall::GetFCallImpl(MethodDesc * pMD, BOOL * pfSharedOrDynamicFCallImpl /*
         return CoreLibBinder::GetMethod(METHOD__DELEGATE__CONSTRUCT_DELEGATE)->GetMultiCallableAddrOfCode();
     }
 
+#ifdef FEATURE_COMINTEROP
     // COM imported classes have special constructors
     if (pMT->IsComObjectType()
```

Comment:
Related: Is there a reason for `IsComObjectType` to ever return true on non-Windows?

Reply:
Not really. That is only set for a `class` if it has been marked with `ComImport`. There is very little the class can do in this case other than exist and be used as metadata.